Lib/http/server.py

```diff
@@ -687,6 +688,21 @@ def send_head(self):
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
         try:
+            # Use browser cache if possible
+            if "If-Modified-Since" in self.headers:
+                # compare If-Modified-Since and date of last file modification
+                fs = os.stat(path)
+                ims = email.utils.parsedate(self.headers["If-Modified-Since"])
+                if ims is not None:
+                    ims_datetime = datetime.datetime(*ims[:7])
+                    ims_dtstring = ims_datetime.strftime("%d %b %Y %H:%M:%S")
+                    last_modif = datetime.datetime.utcfromtimestamp(
+                        fs.st_mtime).strftime("%d %b %Y %H:%M:%S")
+                    if last_modif <= ims_dtstring:
```

Comment:
Why compare string representations rather than datetime objects or just tuples? The order of such strings is different from the order of datetime objects.